### ISSUE TITLE
Wrong executable called in ExecStart line of systemd unit for zabbix-agent2

### ISSUE THREAD

**Issue by amuckart** [voxpupuli/puppet-zabbix#951]
## Affected Puppet, Ruby, OS and module versions/distributions
- Distribution: Ubuntu 22.04
- Module version: 10.0.0

## How to reproduce (e.g Puppet code you use)
Configure agent with hiera
```yaml
---
classes:
  - zabbix::agent

zabbix::agent::agent_configfile_path: '/etc/zabbix/zabbix_agent2.conf'
zabbix::agent::enableremotecommands: 0
zabbix::agent::include_dir: '/etc/zabbix/zabbix_agent2.d'
zabbix::agent::listenip: '127.0.0.1'
zabbix::agent::manage_resources: true
zabbix::agent::manage_startup_script: true
zabbix::agent::monitored_by_proxy: '<proxy-hostname>'
zabbix::agent::server: '127.0.0.1'
zabbix::agent::serveractive: '127.0.0.1'
zabbix::agent::service_enable: true
zabbix::agent::servicename: 'zabbix-agent2'
zabbix::agent::zabbix_package_agent: 'zabbix-agent2'
```
## What are you seeing
Agent fails to start, error in journal shows an attempt to start `/usr/sbin/zabbix_agentd` instead of `/usr/sbin/zabbix_agent2`

## What behaviour did you expect instead
Systemd should start zabbix_agent2

## Output log
```
systemd[822087]: zabbix-agent2.service: Failed to locate executable /usr/sbin/zabbix_agentd: No such file or directory
Subject: Process /usr/sbin/zabbix_agentd could not be executed
Defined-By: systemd
Support: http://www.ubuntu.com/support

The process /usr/sbin/zabbix_agentd could not be executed and failed.

The error number returned by this process is ERRNO.
```

## Any additional information you'd like to impart
File `templates/zabbix-agent-systemd.init.erb` has the executable hard coded:
```erb
ExecStart=/usr/sbin/zabbix_agentd <%= @additional_service_params %> -c <%= @agent_configfile_path %>
```

**Comment by Valantin** [comment]
Hi, as mentioned here https://github.com/voxpupuli/puppet-zabbix#usage-zabbix-agent2 you need to disable manage_startup_script

**Comment by Valantin** [comment]
@amuckart I've migrated 200 node to zabbix-agent2 with the suggested configuration any news about this?

**Comment by amuckart** [comment]
> @amuckart I've migrated 200 node to zabbix-agent2 with the suggested configuration any news about this?

Do you have ‘manage_startup_script’ set to true? 

If not, then the module will leave it alone and use the package provided one which, on Ubuntu at least, is better than the one the module generates and correctly calls zabbix_agent2 not zabbix_agent.

**Comment by Valantin** [comment]
> > @amuckart I've migrated 200 node to zabbix-agent2 with the suggested configuration any news about this?
> 
> Do you have ‘manage_startup_script’ set to true? 
No, as documented for agent2 I've set it to false
> 
> If not, then the module will leave it alone and use the package provided one which, on Ubuntu at least, is better than the one the module generates and correctly calls zabbix_agent2 not zabbix_agent. 

Is the expected behavior because the module provided unit was thinked for old agent version and non needed for agent2

**Comment by Valantin** [comment]
I've add to a PR a note in the readme to clarify manage_startup_script => false was mandatory for agent2

**Comment by amuckart** [comment]
Thank you Valantin. 

I think a cleaner solution would be to either add logic to ensure that if the agent version is 2 that the unit file the module previously installed gets removed, or to add an third option, `purge` to remove the module-created unit file. 

I'm not sure whether making it remove the file when `false` is set would have unintended consequences, but in my case (leaving aside my initial misconfiguration) it would have made the transition easier.

I'm happy to put a PR together to do one of those things.

I did find that for several runs the module was recreating the unit file even when I had set the option to `false` and it wasn't until I toggled `manage_service` to `false` as well that it stopped, but I never got to the bottom of why that was happening.
